modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/LegalEntityDiscountingProvider.java

```diff
@@ -0,0 +1,735 @@
+   * <p>
+   * See {@link LegalEntityDiscountingProvider#repoCurveDiscountFactors(StandardId, StandardId, Currency)}.
+   */
+  @PropertyDefinition(validate = "notNull", get = "private")
+  private final ImmutableMap<StandardId, BondGroup> bondMap;
+  /**
+   * The repo curves, defaulted to an empty map.
+   * The curve data, predicting the future, associated with each bond group and currency.
+   */
+  @PropertyDefinition(validate = "notNull", get = "private")
+  private final ImmutableMap<Pair<BondGroup, Currency>, DiscountFactors> repoCurves;
+
+  /**
+   * The legal entity group map.
+   * <p>
+   * This map is used to convert the {@link StandardId} that identifies the legal entity to
+   * the associated legal entity group in order to lookup an issuer curve.
+   * <p>
+   * See {@link LegalEntityDiscountingProvider#issuerCurveDiscountFactors(StandardId, Currency)}.
+   */
+  @PropertyDefinition(validate = "notEmpty", get = "private")
+  private final ImmutableMap<StandardId, LegalEntityGroup> legalEntityMap;
+  /**
+   * The issuer curves.
+   * The curve data, predicting the future, associated with each legal entity group and currency.
+   */
+  @PropertyDefinition(validate = "notEmpty", get = "private")
+  private final ImmutableMap<Pair<LegalEntityGroup, Currency>, DiscountFactors> issuerCurves;
+
+  //-------------------------------------------------------------------------
+  @ImmutablePreBuild
+  private static void preBuild(Builder builder) {
+    if (builder.valuationDate == null && !builder.issuerCurves.isEmpty()) {
+      builder.valuationDate = builder.issuerCurves.values().iterator().next().getValuationDate();
+    }
+  }
+
+  @ImmutableValidator
+  private void validate() {
+    for (Entry<Pair<BondGroup, Currency>, DiscountFactors> entry : repoCurves.entrySet()) {
+      if (!entry.getValue().getValuationDate().isEqual(valuationDate)) {
+        throw new IllegalArgumentException("Invalid valuation date for the curve: " + entry.getValue().getCurveName());
+      }
+      if (!bondMap.containsValue(entry.getKey().getFirst())) {
+        throw new IllegalArgumentException("No map to the bond group from ID: " + entry.getKey().getFirst());
+      }
+    }
+    for (Entry<Pair<LegalEntityGroup, Currency>, DiscountFactors> entry : issuerCurves.entrySet()) {
+      if (!entry.getValue().getValuationDate().isEqual(valuationDate)) {
+        throw new IllegalArgumentException("Invalid valuation date for the curve: " + entry.getValue().getCurveName());
+      }
+      if (!legalEntityMap.containsValue(entry.getKey().getFirst())) {
+        throw new IllegalArgumentException("No map to the legal entity group from ID: " + entry.getKey().getFirst());
+      }
+    }
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the discount factors of a repo curve for standard IDs and a currency.
+   * <p>
+   * If the valuation date is on or after the specified date, the discount factor is 1.
```

Comment:
Change doc: If the bond standard ID is matched in a BondGroup, the relevant DiscountFactors is returned, if not the issuer standard ID is checked and the relevant DiscountFactors is returned; if both the bond and the issuer ID are not in any BondGroup, an error is thrown.